src/routes/AgreementViewRoute.js

```diff
@@ -226,21 +227,31 @@ class AgreementViewRoute extends React.Component {
       .find(i => i.id === id);
   }
 
-  handleClone = (cloneableProperties) => {
+  handleClone = async (cloneableProperties) => {
     const { history, location, match, stripes: { okapi } } = this.props;
 
-    return fetch(`${okapi.url}/erm/sas/${match.params.id}/clone`, {
+    const response = await fetch(`${okapi.url}/erm/sas/${match.params.id}/clone`, {
       method: 'POST',
       headers: {
         'X-Okapi-Tenant': okapi.tenant,
         'X-Okapi-Token': okapi.token,
         'Content-Type': 'application/json'
       },
       body: JSON.stringify(cloneableProperties),
-    }).then(response => response.json())
-      .then(({ id }) => {
-        history.push(`${urls.agreementEdit(id)}${location.search}`);
-      });
+    });
+
+    try {
+      const text = await response.text(); // Parse it as text
+      const data = JSON.parse(text); // Try to parse it as json
+      if (response.ok) {
```

Comment:
If you're not using JSON_ERROR, you can simplify this section to just `if (response.ok && data.id) ....`